keel-docker/src/main/kotlin/com/netflix/spinnaker/keel/docker/TagVersionStrategy.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ *
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.netflix.spinnaker.keel.docker
+
+import com.netflix.spinnaker.keel.docker.SortType.INCREASING
+import com.netflix.spinnaker.keel.docker.SortType.SEMVER
+
```

Comment:
Here are the options for versioning:
- your tag as-is is either a number (INCREASING_TAG) or semver (SEMVER_TAG), and we do no parsing
- your tag matches one of our defined versioning strategies, and we regex for you
- your tag is custom, so you provide a capture group and a choice of whether that capture group is either a number or a semver

Here are examples for each strategy:
```
 INCREASING_TAG, // tag like : 100
  SEMVER_TAG, // tag like : 1.12.3-rc.1
  BRANCH_JOB_COMMIT_BY_JOB, // popular netflix strategy : master-h1.blah (using the h1)
  SEMVER_JOB_COMMIT_BY_JOB, // popular netflix strategy : v1.12.3-rc.1-h1192.f876e5a (using the h1192)
  SEMVER_JOB_COMMIT_BY_SEMVER; // popular netflix strategy : v1.12.3-rc.1-h1192.f876e5a (using the v1.12.3 blah blah)
```
Naming suggestions welcome, or any other suggestions. I'm trying to make it easy for people to do something netflix endorses while _also_ providing flexibility for defining this any way you want.